fetpype/workflows/utils.py

```diff
@@ -172,23 +172,22 @@ def check_and_update_paths(data_dir, out_dir, nipype_dir, pipeline_name):
     Returns:
         tuple: Updated paths for data_dir, out_dir, and nipype_dir.
     """
+
     data_dir = os.path.abspath(data_dir)
 
+    assert os.path.exists(data_dir), f"Error {data_dir} should be a valid dir"
+
     if out_dir is None:
-        out_dir = os.path.join(data_dir, "derivatives", pipeline_name)
-    else:
-        out_dir = os.path.join(os.path.abspath(out_dir), pipeline_name)
-
-    try:
-        os.makedirs(out_dir)
-    except OSError:
-        print("out_dir {} already exists".format(out_dir))
+        out_dir = data_dir
+
+    out_dir = os.path.join(os.path.abspath(out_dir), "derivatives", pipeline_name)
+
+    os.makedirs(out_dir, exist_ok=True)
+
     if nipype_dir is None:
-        # Get parent directory of data_dir
-        parent_dir = os.path.dirname(data_dir)
-        nipype_dir = os.path.join(parent_dir, "nipype")
-    else:
-        nipype_dir = os.path.abspath(nipype_dir)
+        nipype_dir = out_dir
+
+    nipype_dir = os.path.join(os.path.abspath(nipype_dir), pipeline_name)
```

Comment:
Hey, I think that the interactions are a bit strange when both `out_dir` and `nipype_dir` are None. 

It means that the directories would be: `data_dir/derivatives/pipeline` and `data_dir/derivatives/pipeline/pipeline` if I'm not mistaken. 

What about rather aiming for out_dir=`data_dir/derivatives/pipeline` by default and `nipype_dir=data_dir/derivatives/nipype/pipeline`?